bundle/edu.gemini.itc.web/src/main/java/edu/gemini/itc/web/html/HtmlPrinter.java

```diff
@@ -23,36 +23,32 @@ public static String printParameterSummary(final SourceDefinition sdp) {
-                break;
-            case LIBRARY_STAR:
-                sb.append(" " + sdp.getSourceNormalization() + " " + sdp.units.displayValue() + " " + ((Library) sdp.distribution).sedSpectrum() +
-                        " star in the " + sdp.getNormBand().name() + " band.");
-                break;
-            case LIBRARY_NON_STAR:
-                sb.append(" " + sdp.getSourceNormalization() + " " + sdp.units.displayValue() + " " + ((Library) sdp.distribution).sedSpectrum() +
-                        " in the " + sdp.getNormBand().name() + " band.");
-                break;
-            case USER_DEFINED:
-                sb.append(" a user defined spectrum with the name: " + sdp.getUserDefinedSpectrum());
-                break;
-            case PLAW:
-                sb.append(" Power Law Spectrum, with an index of " + sdp.getPowerLawIndex()
-                        + " and " + sdp.getSourceNormalization() + " mag in the " + sdp.getNormBand().name() + " band.");
-                break;
+        if (sdp.distribution() instanceof EmissionLine) {
+            final EmissionLine eLine = (EmissionLine) sdp.distribution();
+            sb.append(String.format("n emission line, at a wavelength of %.4f microns, ", eLine.wavelength().toMicrons()));
+            sb.append(String.format(
+                    "and with a width of %.2f km/s.\n  It's total flux is %.3e watts_flux on a flat continuum of flux density %.3e watts_fd_wavelength.",
+                    eLine.width().toKilometersPerSecond(), eLine.flux().toWattsPerSquareMeter(), eLine.continuum().toWattsPerSquareMeterPerMicron()));
+        } else if (sdp.distribution() instanceof BlackBody) {
+            final double temperature = ((BlackBody) sdp.distribution()).temperature();
+            sb.append(" " + temperature + "K Blackbody, at " + sdp.norm() +
+                    " " + sdp.units().displayValue() + " in the " + sdp.normBand().name() + " band.");
+        } else if (sdp.distribution() instanceof LibraryStar) {
+            sb.append(" " + sdp.norm() + " " + sdp.units().displayValue() + " " + ((Library) sdp.distribution()).sedSpectrum() +
+                    " star in the " + sdp.normBand().name() + " band.");
+        } else if (sdp.distribution() instanceof LibraryNonStar) {
+            sb.append(" " + sdp.norm() + " " + sdp.units().displayValue() + " " + ((Library) sdp.distribution()).sedSpectrum() +
+                    " in the " + sdp.normBand().name() + " band.");
+        } else if (sdp.distribution() instanceof UserDefinedSpectrum) {
+            sb.append(" a user defined spectrum with the name: " + ((UserDefinedSpectrum) sdp.distribution()).name());
+        } else if (sdp.distribution() instanceof PowerLaw) {
+            final double index = ((PowerLaw) sdp.distribution()).index();
+            sb.append(" Power Law Spectrum, with an index of " + index
+                    + " and " + sdp.norm() + " mag in the " + sdp.normBand().name() + " band.");
```

Comment:
Agreed, by moving all the parameter objects to Scala writing new code became easier, but obviously there is a downside to it on the Java side of things. The question is how much time we are going to spend on this, I was currently not planning on moving all that code to Scala (yet).